### PR TITLE
add recipe for Apple libdispatch (aka Grand Central Dispatch)

### PR DESCRIPTION
**Pull request by fpelliccioni**
Specify library name and version:  **lib/1.0**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/libdispatch/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+from conans.errors import ConanInvalidConfiguration
+import os
+
+class LibDispatchConan(ConanFile):
+    name = "libdispatch"
+    homepage = "https://github.com/apple/swift-corelibs-libdispatch"
+    description = "Grand Central Dispatch (GCD or libdispatch) provides comprehensive support for concurrent code execution on multicore hardware.."
+    topics = ("conan", "libdispatch", "apple", "GCD", "concurrency")
+    url = "https://github.com/conan-io/conan-center-index"
+    license = "MIT"
```
Comment (uilianries) [inline recipes/libdispatch/all/conanfile.py:11]:
```suggestion
    license = "Apache-2.0"
```

File: recipes/libdispatch/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+import os
+
```
Comment (uilianries) [inline recipes/libdispatch/all/conanfile.py:4]:
```suggestion

required_conan_version = ">=1.32.0"

```

File: recipes/libdispatch/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = f'swift-corelibs-{self.name}-swift-{self.version}-RELEASE'
```
Comment (uilianries) [inline recipes/libdispatch/all/conanfile.py:38]:
```suggestion
        extracted_dir = "swift-corelibs-{}-swift-{}-RELEASE".format(self.name, self.version)
```

Python format string is not supported by Python 2.7

**Review by uilianries — APPROVED** [review]

File: recipes/libdispatch/all/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,64 @@
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+import os
+
+required_conan_version = ">=1.32.0"
+
+class LibDispatchConan(ConanFile):
+    name = "libdispatch"
```
Comment (prince-chrismc) [inline recipes/libdispatch/all/conanfile.py:8]: There's a lot of forks of this... perhaps an `apple` prefix needs to be added

Comment (fpelliccioni) [reply recipes/libdispatch/all/conanfile.py:8]: Sorry, forks of what?
libdispatch?

Comment (fpelliccioni) [reply recipes/libdispatch/all/conanfile.py:8]: It is from Apple, you are right, but I think the one from Apple is the original one.

https://github.com/apple/swift-corelibs-libdispatch

Comment (prince-chrismc) [reply recipes/libdispatch/all/conanfile.py:8]: > forks of what? libdispatch?

Yes

I think it was OSS that apple ran with since they needed it

Comment (fpelliccioni) [reply recipes/libdispatch/all/conanfile.py:8]: What I know of is a project run by Apple.
Any variant is a fork of Apple's.

**Review by SSE4 — APPROVED** [review]

**Comment by fpelliccioni** [comment]
Guys, sorry for the inconvenience.
Could you please review this PR?
Thanks a lot!

@madebr @SpaceIm @ericLemanissier @Croydon @intelligide @theirix @gocarlos

**Review by prince-chrismc — APPROVED** [review]

File: recipes/libdispatch/all/conanfile.py [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+    homepage = "https://github.com/apple/swift-corelibs-libdispatch"
+    description = "Grand Central Dispatch (GCD or libdispatch) provides comprehensive support for concurrent code execution on multicore hardware.."
```
Comment (intelligide) [inline recipes/libdispatch/all/conanfile.py:10]:
```suggestion
    description = "Grand Central Dispatch (GCD or libdispatch) provides comprehensive support for concurrent code execution on multicore hardware."
```

File: recipes/libdispatch/all/conanfile.py [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = "swift-corelibs-{}-swift-{}-RELEASE".format(self.name, self.version)
+
```
Comment (intelligide) [inline recipes/libdispatch/all/conanfile.py:41]:
```suggestion
```

**Review by intelligide — CHANGES_REQUESTED** [review]

**Review by intelligide — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]
Try to avoid these editorial changes in the future when there is lots of approvals

**Review by uilianries — APPROVED** [review]

**Pull request merged**